lib/types.js

```diff
@@ -290,7 +295,7 @@ define_type('Decimal32', 0x74);
 define_type('Decimal64', 0x84);
 define_type('Decimal128', 0x94);
 define_type('CharUTF32', 0x73, buffer_uint32be_ops());
-define_type('Timestamp', 0x83, {'write':write_long, 'read':read_long});//TODO: convert to/from Date
+define_type('Timestamp', 0x83, {'write':write_long, 'read':read_timestamp});
```

Comment:
I think your change does make sense. As you also note though it would have the potential to break applications (though we could alter the version accordingly). If I could go back in time, I should have resolved this much earlier. The question I'm wrestling with now is whether the value in fixing this justifies the jump in version, given that it is relatively straightforward to obtain a Date from the currently decoded value...

May I think this over a little? 

Thank you very much anyway for your time on this.

Reply:
Oh, absolutely. This is really just a nicer way to have a conversation :)